server/src/main/java/io/crate/execution/MultiPhaseExecutor.java

```diff
@@ -64,10 +76,14 @@ public static CompletableFuture<SubQueryResults> execute(Map<LogicalPlan, Select
         }
         return CompletableFuture
             .allOf(dependencyFutures.toArray(new CompletableFuture[0]))
-            .thenApply(ignored -> new SubQueryResults(valueBySubQuery));
+            .thenApply(ignored -> {
+                ramAccounting.release();
+                return new SubQueryResults(valueBySubQuery);
+            });
     }
 
-    private static CollectingRowConsumer<?, ?> getConsumer(SelectSymbol.ResultType resultType) {
-        return new CollectingRowConsumer<>(FirstColumnConsumers.getCollector(resultType));
+    private static CollectingRowConsumer<?, ?> getConsumer(SelectSymbol selectSymbol, RamAccounting ramAccounting) {
+        var rowAccounting = new TypedRowAccounting(List.of(selectSymbol.innerType()), ramAccounting);
```

Comment:
Why do we use `selectSymbol.innerType()` here, shouldn't it be `selectSymbol.valueType()`?

Asking because `innerType` would correspond to `ResultType.SINGLE_COLUMN_SINGLE_VALUE` which is the only case where we don't add accouting in this PR, kinda contradiction.

Reply:
No, for the accounting we always need to inner-type because this is the type of the row.

Reply:
`selectSymbol.valueType()` is the array type which we don't want to account on.